tests/integration/pilot/original_src_addr_test.go

```diff
@@ -30,13 +30,24 @@ func TestTproxy(t *testing.T) {
 		Features("traffic.original-source-ip").
 		RequiresSingleCluster().
 		Run(func(ctx framework.TestContext) {
-			workloads, err := apps.PodA[0].Workloads()
-			if err != nil {
-				t.Errorf("failed to get Subsets: %v", err)
-				return
-			}
-			// check the server can see the client's original ip
-			checkOriginalSrcIP(ctx, apps.PodA[0], apps.PodTproxy[0], workloads[0].Address())
+			ctx.NewSubTest("preserve original source ip").Run(func(ctx framework.TestContext) {
+				workloads, err := apps.PodA[0].Workloads()
+				if err != nil {
+					t.Errorf("failed to get Subsets: %v", err)
+					return
+				}
+				// check the server can see the client's original ip
+				checkOriginalSrcIP(ctx, apps.PodA[0], apps.PodTproxy[0], workloads[0].Address())
+			})
+			ctx.NewSubTest("app call itself by pod ip").Run(func(ctx framework.TestContext) {
```

Comment:
nit: should we extend https://github.com/istio/istio/blob/535fc0716e028e8d0292db2b9f5f1a829a64bca8/tests/integration/pilot/common/routing.go#L827 instead? we alreayd have cases for calling the service and localhost, we can just add one checking the pod ip

Reply:
ok, looks reasonable